zaza/openstack/charm_tests/tempest/setup.py

```diff
@@ -0,0 +1,378 @@
+        ctxt['admin_domain_name'] = overcloud_auth['OS_DOMAIN_NAME']
+        ctxt['default_credentials_domain_name'] = (
+            overcloud_auth['OS_PROJECT_DOMAIN_NAME'])
+
+
+def get_tempest_context():
+    """Generate the tempest config context.
+
+    :returns: Context dictionary
+    :rtype: dict
+    """
+    keystone_session = openstack_utils.get_overcloud_keystone_session()
+    ctxt = {}
+    add_application_ips(ctxt)
+    add_nova_config(ctxt, keystone_session)
+    add_neutron_config(ctxt, keystone_session)
+    add_glance_config(ctxt, keystone_session)
+    add_cinder_config(ctxt, keystone_session)
+    add_keystone_config(ctxt, keystone_session)
+    add_environment_var_config(ctxt)
+    add_auth_config(ctxt)
+    return ctxt
+
+
+def render_tempest_config(target_file, ctxt, template):
+    """Render tempest config for specified config file and template.
+
+    :param target_file: Name of file to render config to
+    :type target_file: str
+    :param ctxt: Context dictionary
+    :type ctxt: dict
+    :param template: Template module
+    :type template: module
+    :returns: None
+    :rtype: None
+    """
+    # TODO: switch to jinja2 and generate config based on available services
+    with open(target_file, 'w') as f:
+        f.write(template.file_contents.format(**ctxt))
+
+
+def setup_tempest(tempest_template, accounts_template):
+    """Initialize tempest and render tempest config.
+
+    :param tempest_template: tempest.conf template
+    :type tempest_template: module
+    :param accounts_template: accounts.yaml template
+    :type accounts_template: module
+    :returns: None
+    :rtype: None
+    """
+    if os.path.isdir('tempest-workspace'):
+        try:
+            subprocess.check_call(['tempest', 'workspace', 'remove', '--rmdir',
+                                   '--name', 'tempest-workspace'])
+        except subprocess.CalledProcessError:
+            pass
+    try:
+        subprocess.check_call(['tempest', 'init', 'tempest-workspace'])
+    except subprocess.CalledProcessError:
+        pass
+    render_tempest_config(
+        'tempest-workspace/etc/tempest.conf',
+        get_tempest_context(),
+        tempest_template)
+    render_tempest_config(
+        'tempest-workspace/etc/accounts.yaml',
+        get_tempest_context(),
+        accounts_template)
+
+
+def render_tempest_config_keystone_v2():
+    """Render tempest config for Keystone V2 API.
+
+    :returns: None
+    :rtype: None
+    """
+    setup_tempest(tempest_v2, accounts)
+
+
+def render_tempest_config_keystone_v3():
+    """Render tempest config for Keystone V3 API.
+
+    :returns: None
+    :rtype: None
+    """
+    setup_tempest(tempest_v3, accounts)
+
+
+def add_cirros_alt_image():
+    """Add cirros alternate image to overcloud.
+
+    :returns: None
+    :rtype: None
+    """
+    image_url = openstack_utils.find_cirros_image(arch='x86_64')
+    glance_setup.add_image(
```

Comment:
We should re-use and/or generalize zaza.openstack.charm_tests.glance.setup.add_image